### PR TITLE
cli(asset-saver): end devtoolsLog with a newline

### PR DESCRIPTION
**Pull request by brendankenny**
Inspired by recent discussion about churn in newlines at the end of files updated by `yarn update:sample-artifacts` vs [`.editorconfig`'s `insert_final_newline = true`](https://github.com/GoogleChrome/lighthouse/blob/767036fec01ace00c9e0eafcec824e0e217d0d72/.editorconfig#L9) when editing one of the artifact files manually. The scripted update should match the repo style, just like [we do with `yarn update:sample-json`](https://github.com/GoogleChrome/lighthouse/blob/767036fec01ace00c9e0eafcec824e0e217d0d72/lighthouse-core/scripts/cleanup-LHR-for-diff.js#L49-L50).

Turns out `saveArtifacts()` already saved the main artifacts and trace files with trailing newlines, so only the devtoolsLog files needed updating. `update-report-fixtures.js` also needed tweaking to not second guess `saveArtifacts()`.

### CODE REVIEW

File: lighthouse-core/lib/asset-saver.js [hunk 1]
```diff
@@ -19,6 +19,8 @@ const LHError = require('../lib/lh-error.js');
 // TODO(esmodules): Rollup does not support `promisfy` or `stream.pipeline`. Bundled files
 // don't need anything in this file except for `stringifyReplacer`, so a check for
 // truthiness before using is enough.
+// TODO: Can remove promisify(pipeline) in Node 15.
+// https://nodejs.org/api/stream.html#streams-promises-api
```
Comment (brendankenny) [inline lighthouse-core/lib/asset-saver.js:23]: just moving the TODO to see it more easily with the other hacky workaround that can eventually be removed

File: lighthouse-core/scripts/update-report-fixtures.js [hunk 2]
```diff
@@ -37,32 +34,30 @@ async function update(artifactName) {
   await cli.runLighthouse(url, flags, budgetedConfig);
   await server.close();
 
+  let newArtifacts = assetSaver.loadArtifacts(artifactPath);
+
+  // Normalize some data so it doesn't change on every update.
```
Comment (brendankenny) [inline lighthouse-core/scripts/update-report-fixtures.js:39]: this file was making new artifacts, saving them with `saveArtifacts()`, loading them again to normalize churn-y `Timing` then doing it's own `fs.writeFileSync()`. Normalization should just happen before the first (and only) save, which also means the concluding newline added by `saveArtifacts()` will be in the resulting file.

File: lighthouse-core/test/lib/asset-saver-test.js [hunk 3]
```diff
@@ -96,6 +96,7 @@ describe('asset-saver helper', () => {
           {args: {IsMainFrame: true}, cat: 'v8', pid: 1, ts: 5},
           {args: {data: {encodedDataLength: 20, requestId: '1.22'}}, pid: 1, ts: 6},
         ],
+        metadata: {'cpu-model': 9001, 'network-type': 'Unknown'},
```
Comment (brendankenny) [inline lighthouse-core/test/lib/asset-saver-test.js:99]: not directly important to Lighthouse, but overlooked in previous testing of `saveTrace()` and very relevant to where the various methods are putting newlines within their output :)

File: lighthouse-core/lib/asset-saver.js [hunk 4]
```diff
@@ -250,10 +250,12 @@ async function saveTrace(traceData, traceFilename) {
  * @return {Promise<void>}
  */
 function saveDevtoolsLog(devtoolsLog, devtoolLogFilename) {
-  const logIter = arrayOfObjectsJsonGenerator(devtoolsLog);
   const writeStream = fs.createWriteStream(devtoolLogFilename);
 
-  return pipeline(logIter, writeStream);
+  return pipeline(function* () {
+    yield* arrayOfObjectsJsonGenerator(devtoolsLog);
+    yield '\n';
```
Comment (adamraine) [inline lighthouse-core/lib/asset-saver.js:257]: First time seeing `yield` in JS, lemme just google that quick

**Review by adamraine — APPROVED** [review]

**Pull request merged**